packages/common/src/search/wellKnownCatalog.ts

```diff
@@ -0,0 +1,268 @@
+import { IUser } from "@esri/arcgis-rest-types";
+import { getFamilyTypes } from "../content";
+import { EntityType, IHubCatalog, IHubCollection } from "./types";
+
+export type WellKnownCatalog =
+  | "myContent"
+  | "myGroup"
+  | "favorites"
+  | "organization"
+  | "world";
+
+export type WellKnownCollection =
+  | "appAndMap"
+  | "content"
+  | "dataset"
+  | "document"
+  | "event"
+  | "feedback"
+  | "initiative"
+  | "project"
+  | "site"
+  | "solution"
+  | "template";
+
+/** This function returns a single IHubCatalog definition JSON object based on the
+ * name and entity type requested
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param entityType
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ */
+export function getWellKnownCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  entityType: EntityType,
+  user?: IUser,
+  collectionNames?: WellKnownCollection[]
+): IHubCatalog {
+  switch (entityType) {
+    case "item":
+      return getWellknownItemCatalog(i18nScope, name, user, collectionNames);
+    /* Add other entity handlers here, e.g. getWellknownEventCatalog */
+    default:
+      throw new Error(`Wellknown catalog not implemented for "${entityType}"`);
+  }
+}
+
+/**
+ * @param i18nScope Translation scope to be interpolated into the catalog
+ * @param name Name of the catalog requested
+ * @param user Owner of the entity, optional but required for certain catalogs
+ * @param collectionNames A list of collection names, optional, if passed in,
+ * only those collections in the catalog will be returned
+ * @returns An Item IHubCatalog definition JSON object
+ */
+function getWellknownItemCatalog(
+  i18nScope: string,
+  name: WellKnownCatalog,
+  user?: IUser,
+  collectionNames?: WellKnownCollection[]
+): IHubCatalog {
+  // Check if i18nScope is defined, if so add a `.` on the end
+  if (i18nScope !== "") {
+    i18nScope = `${i18nScope}.`;
+  }
+  let catalog;
+  switch (name) {
+    case "myContent":
+      if (user) {
+        catalog = {
+          schemaVersion: 1,
+          title: `{{${i18nScope}catalog.myContent:translate}}`,
+          scopes: {
+            item: {
+              targetEntity: "item" as EntityType,
+              filters: [{ predicates: [{ owner: user.username }] }],
+            },
+          },
+          collections: getWellknownCollections(
```

Comment:
If you don't want to repeat the `getWellKnownCollections` logic every single time, you could bring the initialization outside the switch statement.